sfge/src/main/java/com/salesforce/rules/ops/visitor/LoopDetectionVisitor.java

```diff
@@ -129,7 +189,7 @@ public void afterVisit(BaseSFVertex vertex, SymbolProvider symbols) {
         final List<String> vertexEndScopes = vertex.getEndScopes();
 
         // Look at the list in reverse order to get the newest innermost scope first.
-        for (int i = vertexEndScopes.size() - 1; i >= 0; i--) {
+        for (int i = 0; i < vertexEndScopes.size(); i++) {
```

Comment:
We can talk it through on Monday, but we should probably add a test to validate that the scopes are consistently added in the order we think they are, if we don't already have one.

Reply:
Fixed in 306ae93.